worlds/sc2/client.py

```diff
@@ -909,6 +956,188 @@ def uncollected_locations_in_mission(self, mission: SC2Mission) -> typing.Iterab
 
     def is_mission_completed(self, mission_id: int) -> bool:
         return get_location_id(mission_id, 0) in self.checked_locations
+    
+        
+    async def trade_acquire_storage(self, keep_trying: bool = False) -> typing.Optional[dict]:
+        # This function was largely taken from the Pokemon Emerald client
+        """
+        Acquires a lock on the Void Trade DataStorage.
+        Locking the key means you have exclusive access
+        to modifying the value until you unlock it or the key expires (5 seconds).
+
+        If `keep_trying` is `True`, it will keep trying to acquire the lock
+        until successful. Otherwise it will return `None` if it fails to
+        acquire the lock.
+        """
+        while not self.exit_event.is_set() and self.last_bot and self.last_bot.game_running:
+            lock = int(time.time_ns() / 1000000)
+
+            # Make sure we're not past the waiting limit
+            # SC2 needs to be notified within 10 minutes (training time of the dummy units)
```

Comment:
I've done this and also changed it to use the bot's time measurement, which makes the wait check resilient against game pauses, because the bot calculates time directly from the number of observed game loops. I looked to see if there's a way to react to `speed` commands too, but couldn't find anything.

Reply:
Speed is a debug command but standard game speeds need to be supported